extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -112,14 +112,25 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 	private readonly _requiredKernelPackages: Map<string, PythonPkgDetails[]>;
 	private readonly _requiredPackagesSet: Set<string>;
 
-	constructor(extensionPath: string, outputChannel: OutputChannel, apiWrapper: ApiWrapper, pythonInstallationPath?: string) {
+	public readonly runningOnSAW: boolean;
+
+	constructor(extensionPath: string, outputChannel: vscode.OutputChannel, apiWrapper: ApiWrapper) {
 		this.extensionPath = extensionPath;
 		this.outputChannel = outputChannel;
 		this.apiWrapper = apiWrapper;
-		this._pythonInstallationPath = pythonInstallationPath || JupyterServerInstallation.getPythonInstallPath(this.apiWrapper);
+
+		this.runningOnSAW = vscode.env.appName.toLowerCase().indexOf('saw') > 0;
+		this.apiWrapper.setCommandContext('notebook:runningOnSAW', this.runningOnSAW);
+
+		if (this.runningOnSAW) {
```

Comment:
What about having this be a more generic thing that can be toggled from a set of conditions instead of just saw? Basically - do you see any value in allowing this to be a setting that can be enabled in other conditions?

Reply:
Currently we're only using this prepackaged behavior on SAWs. If we want to expand it to anything else, then we can update it at that time.